### ISSUE TITLE
Problem with get_interp_val and get_neighbours

### ISSUE THREAD

**Issue by sbenton** [healpy/healpy#239]
I'm having a problem with `get_interp_val` producing horizontally extended structures. This appears to be due to the underlying `Healpix_Base::get_interpol` function also used in `get_neighbours`. For example:

```python
import healpy as hp
import numpy as np
pix, wgt = hp.get_neighbours(2, np.arccos(0.0), np.pi*1.0, nest=False)
print(pix)
print(wgt)
```
Which produces:
```
[23 24 32 33]
[ 0.5  0.5  0.   0. ]
```

I compare the pixel numbers to those in Figure 3 of: http://healpix.sourceforge.net/html/intronode4.htm in the top panel. With the test point at the intersection of the boundaries of four pixels, I expect the result to be pixels [16 23 24 32] with similar weights (possibly different for the vertical/horizontal pairs). However, `get_neighbours` returns pixel 33 rather than 16 and assigns weight of 0.0 to pixels 32 and 33. Moving up and down a little in theta, I can get a non-zero weight for one vertical neighbour, but still get a more distant neighbour with weight 0.0

Changing to `nest=True` yields equivalent results.

**Comment by mreineck** [comment]
The problem here is the misleading name of the healpy function get_neighbours(),because it does not actually compute any neighbours ...
The underlying C++ function get_interpol does the following:
- first, it determines the ring of pixels above and below the theta of the provided location ... this already is ambiguous, since you are specifying a theta which lies exactly on a ring. But for the purposes of the get_interpol routine this doen't matter.
- then it determines the pixels closest to the provided phi parameter on both of the rings found above.
- Finally, it computes weights for linear interpolation for these four pixels and returns everything.

I'm pretty sure that the C++ part behaves exactly as it should. The Python function should be renamed, I guess. Certainly its documentation should be fixed, because get_interpol definitely does not return the four closest pixels.

**Comment by sbenton** [comment]
Thanks, mreineck, for your explanation.

I think the problem is that `get_interpol` selects pixels along two rings in cases where the nearest pixels lie along three rings. This favours ring-aligned structures and results in subtle stripes in my map, interpolated with `get_interp_val`.

**Comment by mreineck** [comment]
The choice of using bilinear interpolation along longitude and latitude was made after quite some discussion, when it was introduced. At the time it was agreed that this approach made more physical sense that one based on the nearest pixel centers.
It is true that this approach will produce stripes, but at the same time, using the four nearest pixels will produce diamond-shaped artifacts; this is unavoidable with any kind of bilinear interpolation.
A further complication of the interpolation based on nearest points is that in some regions on the sphere, the interpolation should not be done between four pixels, but in fact between only three - if you look at the Healpix pixelization for Nside=1 you will see these regions immediately.
In short, we are pretty sure that switching to another interpolation scheme will make things more complicated for no real practical gain.

**Comment by sbenton** [comment]
That makes sense.

I agree that updating documentation will at least help prevent people like me from getting confused in the future.

Thanks again.
